File: summer/WebContent/js/MS/internal/data/AccessorInfo.js
/**
 * from AccessorTable
 * AccessorInfo
 */

define(["dojo/_base/declare", "system/Type"], 
		function(declare, Type){
	var AccessorInfo = declare("AccessorInfo", Object,{
		constructor:function(/*object*/ accessor, /*Type*/ propertyType, /*object[]*/ args)
        { 
            this._accessor = accessor;
            this._propertyType = propertyType; 
            this._args = args; 
            
//            int         
            this._generation = 0;        // used for discarding aged entries
		},
		
	});
	
	Object.defineProperties(AccessorInfo.prototype,{
//	     internal object     
		Accessor:        { get:function() { return this._accessor; } },
//        internal Type       
		PropertyType:    { get:function() { return this._propertyType; } },
//        internal object[]   
		Args:            { get:function() { return this._args; } },
	 
//        internal int 
		Generation: { get:function() { return this._generation; }, set:function(value) { this._generation = value; } }	  
	});
	
	Object.defineProperties(AccessorInfo,{
		  
	});
	
	AccessorInfo.Type = new Type("AccessorInfo", AccessorInfo, [Object.Type]);
	return AccessorInfo;
});

File: org.summer.view.widget/src/org/summer/view/widget/data/AccessorInfo.java
package org.summer.view.widget.data;

import org.summer.view.widget.Type;
/*internal*/ public /*sealed*/ class AccessorInfo
    {
        /*internal*/ public AccessorInfo(Object accessor, Type propertyType, Object[] args)
        { 
            _accessor = accessor;
            _propertyType = propertyType; 
            _args = args; 
        }
 
        /*internal*/ public Object     Accessor        { get { return _accessor; } }
        /*internal*/ public Type       PropertyType    { get { return _propertyType; } }
        /*internal*/ public Object[]   Args            { get { return _args; } }
 
        /*internal*/ public int Generation { get { return _generation; } set { _generation = value; } }
 
        Object      _accessor;          // DP, PD, or PI 
        Type        _propertyType;      // type of the property
        Object[]    _args;              // args for indexed property 
        int         _generation;        // used for discarding aged entries
    }
